UPGRADE-4.0.md

```diff
@@ -758,6 +758,9 @@ Security
 
  * The `GuardAuthenticatorInterface` interface has been removed.
    Use `AuthenticatorInterface` instead.
+   
+ * When extending `AbstractGuardAuthenticator` it's deprecated to return `null` from `getCredentials()`.
```

Comment:
In 4.0, it's not deprecated anymore. So, it should probably be something like "When extending `AbstractGuardAuthenticator`,  `getCredentials()` cannot return `null` anymore, return `false` from `supports()` if no credentials available instead."